include/ensmallen_bits/aug_lagrangian/aug_lagrangian.hpp

```diff
@@ -49,13 +48,24 @@ class AugLagrangian
    *     class.
    * @param function The function to optimize.
    * @param coordinates Output matrix to store the optimized coordinates in.
+   * @param penaltyThresholdFactor When the penalty threshold is updated set
+   *    the penalty threshold to the penalty multplied by this factor. The
+   *    default value of 0.25 is is taken from Burer and Monteiro (2002).
+   * @param sigmaUpdateFactor When sigma is updated  multiply sigma by this
+   *    value. The default value of 10 is taken from Burer and Monteiro (2002).
    * @param maxIterations Maximum number of iterations of the Augmented
    *     Lagrangian algorithm.  0 indicates no maximum.
+   * @param internalMaxIterations Maximum number of iterations of L-BFGS
+   *    iterations internal to the Augmented Lagrangian algorithm.
+   *    0 indicates no maximum.
    */
   template<typename LagrangianFunctionType>
   bool Optimize(LagrangianFunctionType& function,
                 arma::mat& coordinates,
-                const size_t maxIterations = 1000);
+                const double penaltyThresholdFactor = 0.25,
```

Comment:
Nice catch, I think everything should go into the class constructor, to be consistent with the other optimizers. Let me know what you think.

Reply:
Yeah makes sense, I'll move the algorithm arguments across (i.e. maxIterations, defaultMaxIterations, penaltyThresholdFactor and sigmaUpdateFactor).

Happy with the initSigma and initLambda being in the function call still rather than the constructor?

Reply:
We can move the parameter as well, and provide a function for each to get/modify the value.